### ISSUE TITLE
SA1118 still issues warning eventhough disabled in *.ruleset

### ISSUE THREAD

**Issue by ltvan** [DotNetAnalyzers/StyleCopAnalyzers#2647]
Environment: VS2017, StyleCop.Analyzers 1.0.2.

I disabled SA1118 (None or Hidden), the editor doesn't show warning but running build still issues warning in this case:

```csharp
        public static readonly EmailTemplateDefinition TestEmail = new EmailTemplateDefinition(
            name: "app.email-templates.test-email",
            subject: "Test email ignore please",
            body: @"<p>This is test email, please ignore.</p>
<p></p>
<p>---</p>
<p>FxCompanyName</p>",
            editable: false);
```

Calling inside another method doesn't produce any warning.

Disable by `pragma` doesn't work too. Workaround using `SuppressMessage` ok.

**Comment by sharwell** [comment]
> I disabled SA1118 (None or Hidden) ... but running build still issues warning ...

This type of suppression is handled by the compiler core and we couldn't override it even if we wanted to. The only way the warning could occur is if the ruleset isn't actually getting used by the build (e.g. if the property is overridden to pass a different rule set file, or if the path to the file is incorrect on the system where the build is running).

**Comment by ltvan** [comment]
I can be sure the ruleset file is applied consistantly, since other rules are fine, and SA1118 is also fine in other situations. It only occurs in the `static readonly` field initialization.

**Comment by ltvan** [comment]
An additional note is that the "build" I mentioned is the Build command in Visual Studio, not the CI build.

**Comment by sharwell** [comment]
If you are able to post a sample project to GitHub, I may be able to find the issue. :) Otherwise, a copy of the project file may contain enough information (depending on customizations you may have).

**Comment by ltvan** [comment]
Sorry, I think I found the problem. I migrated my project from Stylecop.MSBuild.Targets but forgot to remove the link to the target file. So that warning is from the old one. When I tried in other situations, I placed the long params at the first parameter so it passed the rule.

PS: the old Stylecop rule statement is missing this part `other than the first parameter` so I didn't recognize that:

```
warning : SA1118 : CSharp.Readability : The parameter spans multiple lines. If the parameter is short, place the entire parameter on a single line. Otherwise, save the contents of the parameter in a temporary variable and pass the temporary variable as a parameter. 
```

**Comment by sharwell** [comment]
:+1: glad to hear it's worked out!
